infrastructure/templates/infrastructure/powerplant_detail.html

```diff
@@ -132,6 +145,23 @@ <h3>CO2 Emissions</h3>
       </div>
     </section>
 
+    <section class="related-power-data">
+        <h2>Associated Project Initiatives</h2>
+        
+        {% for project in object.project_set.published %}
```

Comment:
I think you could cut down on the query count a bit by performing a more specific select within the view. For example:

```python
projects_with_initiatives = self.object.project_set.exclude(initiatives__isnull=True)
context['project_initiatives'] = projects_with_initiatives.prefetch_related('initiatives')
```

Reply:
Actually, upon further reading of the ticket re _Duplicates should be supressed_ , this'll need to be improved first.

Reply:
Probably easiest to come from Initiative, maybe something like:

```python
Initiative.objects.filter(project__power_plant_id=self.object.pk,
                          project__published=True).distinct().order_by('project__pk')
```

Reply:
The `distinct` here doesn't seem to work for me. I've set up a situation where i have a powerplant that assigned to 3 different projects and those projects have initiatives. Two of the projects have the "Karakoram Highway" as an Initiative.  I get this back:

```
Out[22]: <TreeQuerySet [<Initiative: Afghanistan National Development Strategy>, <Initiative: Karakoram Highway>, <Initiative: South Asia Subregional Economic Cooperation (SASEC)>, <Initiative: Belt and Road>, <Initiative: China-Pakistan Economic Corridor (CPEC)>, <Initiative: Karakoram Highway>]>
```

The query works if we leave out the `order_by`, but then we run into the same complications with grouping and display.

Reply:
The below commits are a bit solve this issue by choosing not to identify the project that the related initiative is attached to. Instead we just list "plant initiatives" and "associated project initiatives." Moved these list up to the top of the page.